### PR TITLE
Refactor to load projects from database

### PR DESCRIPTION
**Pull request by jeanpimentel**

### CODE REVIEW

File: app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionActivity.kt [hunk 1]
```diff
@@ -11,33 +11,34 @@ import android.view.View.INVISIBLE
...
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_model_selection)
+        setContentView(R.layout.activity_project_selection)
 
-        startService(RemoteModelService.createIntent(this))
-        
-        presenter = ModelSelectionPresenter(ModelSelectionInteractor())
+        startService(RemoteProjectService.createIntent(this))
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionActivity.kt]: What do you think of adding it to IntentHelper as well?

File: app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    private class LoadProjectsTask(context: Context, private var onSuccess: (List<Project>) -> Unit, private var onFailure: (Exception) -> Unit) : AsyncTask<Void, Void, LoadProjectsResult>() {
+
+        private val repository = ProjectRepository(context)
+
+        override fun doInBackground(vararg params: Void?): LoadProjectsResult {
+            try {
+                // REMOVE ON SMTC-1488
+                val projects = FakeProjectGenerator.getProjects()
```
Comment (ivamluz) [inline app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionInteractor.kt]: Is this intended?

Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionInteractor.kt]: No, I pushed a commit that I didn't want. Fixed

File: app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionActivity.kt (hunk 1 above)
Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionActivity.kt]: Good idea

Comment (jeanpimentel) [reply app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionActivity.kt]: What do you think about:

`RemoteProjectService.start(this)` and `RemoteProjectService.stop(this)`

Start/Stop methods in Companion object like this:
```
fun start(context: Context) {
     val intent = Intent(context, RemoteProjectService::class.java)
     context.startService(intent)
 }
```

Comment (ivamluz) [reply app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionActivity.kt]: Sounds much better.

**Review by ivamluz — APPROVED** [review]

**Pull request merged**